spec/rspec/core/example_group_spec.rb

```diff
@@ -140,6 +140,12 @@ def metadata_hash(*args)
         expect(child).to have_class_const("SomeParentGroup::Hash")
       end
 
+      it 'removes it\'s constants' do
+        groups = RSpec::ExampleGroups
+        groups.remove_all_constants
```

Comment:
I implemented this test because I didn't know how to verify if an object had been garbage collected. 

How would this test change if I was to put it in the test's you have identified?